### PR TITLE
feat(data): appsync apollo extensions

### PR DESCRIPTION
**Pull request by lawmicha**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [x] Swift
- [x] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+export function getStaticProps(context) {
+  return {
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+AWS AppSync Apollo Extensions provides a seamless way to connect to your AWS AppSync while using the Apollo client. Apollo client is an open-source GraphQL client.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:33]: can we add a link to "Apollo client"?

Comment (josefaidt) [reply src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:33]: disregard I see the following notes, that looks good

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    'android',
+    'angular',
+    'flutter',
+    'javascript',
+    'nextjs',
+    'react',
+    'react-native',
+    'swift',
+    'vue'
+  ]
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:17]:
```suggestion
  platforms: [
    'android',
    'swift',
  ]
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+Enter its GitHub URL (`https://github.com/aws-amplify/aws-appsync-apollo-extensions-swift`), select **Up to Next Major Version** and click **Add Package**
+
+* Select the following libraries:
+    * **AWSAppSyncApolloExtensions**
+
+</InlineFilter>
+
+### Connecting to AWS AppSync with Apollo client
+
+AWS AppSync supports the following authorization modes (https://docs.aws.amazon.com/appsync/latest/devguide/security-authz.html)
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:69]:
```suggestion
AWS AppSync supports the following [authorization modes](https://docs.aws.amazon.com/appsync/latest/devguide/security-authz.html)
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+
+You can provide your own custom `fetchLatestAuthToken` provider for **AWS_LAMBDA** and **OPENID_CONNECT** auth modes. 
+
+#### AWS_IAM
+
+<InlineFilter filters={["swift"]}>
+
+If you are using Amplify Auth, you can use the following method for AWS_IAM auth
+
+```
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:144]:
````suggestion
```swift
````

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+        region: "[REGION]"))
+```
+
+</InlineFilter>
+
+<InlineFilter filters={["android"]}>
+
+If you are using Amplify Auth, you can use the following method for AWS_IAM auth
+
+```
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:159]:
````suggestion
```kotlin
````

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+val authorizer = IamAuthorizer { ApolloAmplifyConnector.signAppSyncRequest(it, "us-east-1") }
+```
+
+</InlineFilter>
+
+### Connecting Amplify Data to Apollo client
+
+Before you begin, you will need an Amplify Data backend deploy. To get started, see [Set up Data](/[platform]/swift/build-a-backend/data/set-up-data/). 
+
+Once you have deployed your backend and created the `amplify_outputs.json`, you can use Amplify library to read and retrieve your configuration values with the following steps:
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:169]:
```suggestion
Once you have deployed your backend and created the `amplify_outputs.json` file, you can use Amplify library to read and retrieve your configuration values with the following steps:
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+*  `owner` strategy, `userPool` authMode, **AuthTokenAuthorizer**
+
+If you define multiple authorization strategies on a single model, you will have to create separate Apollo client instances for each Authorizer that you want to use in your app.
+
+### Downloading the AWS AppSync schema
+
+The schema is used by Apollo’s code generation tool to generate API code that helps you execute GraphQL operations. To retrieve your AWS AppSync schema:
+
+<InlineFilter filters={["swift"]}>
+1. Navigate to your API on the AWS AppSync console
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:224]:
```suggestion
1. Navigate to your API on the [AWS AppSync console](https://console.aws.amazon.com/appsync/home)
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+2. On the left side, select Schema
+3. When viewing your schema, there should a “Export schema” drop down. Select this and download the `schema.json` file.
+4. Add this file to your project as directed by [Apollo Code Generation documentation](https://www.apollographql.com/docs/ios/code-generation/introduction) 
+</InlineFilter>
+
+<InlineFilter filters={["android"]}>
+1. Navigate to your API on the AWS AppSync console
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:231]:
```suggestion
1. Navigate to your API on the [AWS AppSync console](https://console.aws.amazon.com/appsync/home)
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+
+*  `publicAPIkey` strategy, `apiKey` authMode, **APIKeyAuthorizer**
+*  `guest` strategy, `identityPool` authMode, **IAMAuthorizer**
+*  `owner` strategy, `userPool` authMode, **AuthTokenAuthorizer**
+
+If you define multiple authorization strategies on a single model, you will have to create separate Apollo client instances for each Authorizer that you want to use in your app.
+
+### Downloading the AWS AppSync schema
+
+The schema is used by Apollo’s code generation tool to generate API code that helps you execute GraphQL operations. To retrieve your AWS AppSync schema:
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:221]: I wonder if we should include a programmatic example (despite not having a command to do this in the backend cli). In development these schema's will change quite often and downloading each time is cumbersome

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+4. Add this file to your project as directed by [Apollo Code Generation documentation](https://www.apollographql.com/docs/ios/code-generation/introduction) 
+</InlineFilter>
+
+## Connecting to AWS AppSync real-time endpoint
+
+The following example shows how you can create an Apollo client that allows performing GraphQL subscription operations with AWS AppSync. 
+
+<InlineFilter filters={["swift"]}>
+
+```
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:243]:
````suggestion
```swift
````

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx (hunk 9 above)
Comment (lawmicha) [reply src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:214]: i think the base example should exist, as a way to show a simple way to get the schema from the console, while your suggestion to extend the example makes sense. The ideal example should utilize Apollo's codegen configuration to provide it the introspection endpoint and let Apollo retrieve it as part of the codegen process. 

Other programmatic examples would involve taking on a dependency on AWS SDK, like for Javascript and writing a script to download the schema using the appsync client

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 11]
```diff
@@ -0,0 +1,287 @@
...
+
+</InlineFilter>
+
+### Features
+
+AWS AppSync Apollo Extensions provide AWS AppSync authorizers to be used with the Apollo client to make it simple to configure runtime interceptors and apply the correct authorization payloads to your GraphQL operations.
+
+The Amplify library provides components to facilitate configuring the authorizers with Apollo client by providing configuration values to connect to your Amplify Data backend.
+
+<InlineFilter filters={["swift"]}>
```
Comment (mattcreaser) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:47]: Need Android version of this section.

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 12]
```diff
@@ -0,0 +1,287 @@
...
+
+</InlineFilter>
+
+### Connecting Amplify Data to Apollo client
+
+Before you begin, you will need an Amplify Data backend deploy. To get started, see [Set up Data](/[platform]/swift/build-a-backend/data/set-up-data/). 
+
+Once you have deployed your backend and created the `amplify_outputs.json` file, you can use Amplify library to read and retrieve your configuration values with the following steps:
+
+<InlineFilter filters={["swift"]}>
```
Comment (mattcreaser) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:164]: Need Android version of this section

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 13]
```diff
@@ -0,0 +1,287 @@
...
+2. On the left side, select Schema
+3. When viewing your schema, there should a “Export schema” drop down. Select this and download the `schema.json` file.
+4. Add this file to your project as directed by [Apollo Code Generation documentation](https://www.apollographql.com/docs/ios/code-generation/introduction) 
+</InlineFilter>
+
+<InlineFilter filters={["android"]}>
+1. Navigate to your API on the [AWS AppSync console](https://console.aws.amazon.com/appsync/home)
+2. On the left side, select Schema
+3. When viewing your schema, there should a “Export schema” drop down. Select this and download the `schema.json` file.
+4. Add this file to your project as directed by [Apollo Code Generation documentation](https://www.apollographql.com/docs/ios/code-generation/introduction)
```
Comment (mattcreaser) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:227]:
```suggestion
4. Add this file to your project as directed by [Apollo documentation](https://www.apollographql.com/docs/kotlin/advanced/plugin-recipes#specifying-the-schema-location) 
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 14]
```diff
@@ -0,0 +1,287 @@
...
+```kotlin
+// Create the network transport
+val networkTransport = WebSocketNetworkTransport.Builder()
+    .protocol(AppSyncProtocol.Factory(endpoint, authorizer))
+    .build()
+
+// Use the network transport when creating the Apollo Client
+val apolloClient = ApolloClient.Builder()
+    .subscriptionNetworkTransport(networkTransport)
+    .build()
```
Comment (mattcreaser) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:284]:
```suggestion
val endpoint = AppSyncEndpoint("<your_appsync_endpoint>")
val authorizer = /* your Authorizer */

val apolloClient = ApolloClient.Builder()
    .appSync(endpoint, authorizer)
    .build()
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx (hunk 9 above)
Comment (lawmicha) [reply src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx]: I added an example for swift for fetching the introspection schema programmatically using `apollo-ios-cli`

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx (hunk 11 above)
Comment (lawmicha) [reply src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:63]: Added android section, can you confirm the name? I have set it roughly as `com.amplifyframework:aws-appsync-apollo-extensions-android:1.0.0`

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx (hunk 12 above)
Comment (lawmicha) [reply src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:180]: Added the Android section, which is 
1. the gradle for the second package, could you update to the final name?
2. a code example , need to add this

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 15]
```diff
@@ -0,0 +1,321 @@
...
+<InlineFilter filters={["android"]}>
+
+### Install the AWS AppSync Apollo library
+
+Add AWS AppSync Apollo Extensions dependency to your app/build.gradle.kts file:
+
+```kotlin title="app/build.gradle.kts"
+dependencies {
+    // highlight-start
+    implementation("com.amplifyframework:aws-appsync-apollo-extensions-android:1.0.0")
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:56]:
```suggestion
    implementation("com.amplifyframework:apollo-appsync:1.0.0")    
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx (hunk 11 above)
Comment (lawmicha) [reply src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:63]: Updated

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 16]
```diff
@@ -0,0 +1,321 @@
...
+</InlineFilter>
+
+<InlineFilter filters={["android"]}>
+
+Add Amplify AppSync Extensions dependency to your app/build.gradle.kts file:
+
+```kotlin title="app/build.gradle.kts"
+dependencies {
+    // highlight-start
+    implementation("com.amplifyframework:amplify-appsync-apollo:1.0.0")
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:224]:
```suggestion
    implementation("com.amplifyframework:apollo-appsync-amplify:1.0.0")    
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx (hunk 12 above)
Comment (lawmicha) [reply src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:180]: Updated dependency name

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 17]
```diff
@@ -0,0 +1,321 @@
...
+
+    return ApolloClient(networkTransport: transport, store: store)
+}
+```
+
+</InlineFilter>
+
+<InlineFilter filters={["android"]}>
+
+Add Amplify AppSync Extensions dependency to your app/build.gradle.kts file:
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:219]:
```suggestion
Add `apollo-appsync-amplify` dependency to your app/build.gradle.kts file:
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 18]
```diff
@@ -0,0 +1,321 @@
...
+
+AWS AppSync Apollo Extensions provide AWS AppSync authorizers to be used with the Apollo client to make it simple to configure runtime interceptors and apply the correct authorization payloads to your GraphQL operations.
+
+The Amplify library provides components to facilitate configuring the authorizers with Apollo client by providing configuration values to connect to your Amplify Data backend.
+
+<InlineFilter filters={["android"]}>
+
+### Install the AWS AppSync Apollo library
+
+Add AWS AppSync Apollo Extensions dependency to your app/build.gradle.kts file:
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:51]:
```suggestion
Add `apollo-appsync` dependency to your app/build.gradle.kts file:
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 19]
```diff
@@ -0,0 +1,321 @@
...
+```kotlin title="app/build.gradle.kts"
+dependencies {
+    // highlight-start
+    implementation("com.amplifyframework:apollo-appsync-amplify:1.0.0")    
+    // highlight-end
+}
+```
+
+```kotlin
+Code Sample
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:230]:
```suggestion
// Use CUP auth mode, reading configuration from AmplifyOutputs
val connector = ApolloAmplifyConnector(context, AmplifyOutputs(R.raw.amplify_outputs))
val apolloClient = ApolloClient.Builder()
    .appSync(connector.endpoint, connector.apiKeyAuthorizer())
    .build()
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 20]
```diff
@@ -0,0 +1,325 @@
...
+```kotlin title="app/build.gradle.kts"
+dependencies {
+    // highlight-start
+    implementation("com.amplifyframework:apollo-appsync-amplify:1.0.0")    
+    // highlight-end
+}
+```
+
+```kotlin
+// Use CUP auth mode, reading configuration from AmplifyOutputs
```
Comment (lawmicha) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:230]:
```suggestion
// Use apiKey auth mode, reading configuration from AmplifyOutputs
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx (hunk 12 above)
Comment (lawmicha) [reply src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:180]: Updated code snippet

**Review by josefaidt — DISMISSED** [review]

**Review by josefaidt — DISMISSED** [review]

**Review by josefaidt — APPROVED** [review]

**Review by mattcreaser — APPROVED** [review]

**Review by vincetran — APPROVED** [review]

**Review by thisisabhash — APPROVED** [review]

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 21]
```diff
@@ -0,0 +1,398 @@
...
+
+To learn more about Apollo, see https://www.apollographql.com/docs/kotlin.
+
+</InlineFilter>
+
+## Features
+
+AWS AppSync Apollo Extensions provide AWS AppSync authorizers to be used with the Apollo client to make it simple to apply the correct authorization payloads to your GraphQL operations.
+
+The Amplify library provides components to facilitate configuring the authorizers with Apollo client by providing configuration values to connect to your Amplify Data backend.
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:45]:
```suggestion
The Amplify library provides components to facilitate configuring the authorizers with Apollo client by providing configuration values to connect to your Amplify data backend.
```

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 22]
```diff
@@ -0,0 +1,398 @@
...
+export function getStaticProps(context) {
+  return {
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+AWS AppSync Apollo Extensions provide a seamless way to connect to your AWS AppSync backend using Apollo client, an open-source GraphQL client.
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:26]: can you find and replace lowercase e for Extensions in this - this isn't an official name so needs to be lowercase (at least I haven't seen the naming doc :))

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 23]
```diff
@@ -0,0 +1,398 @@
...
+//highlight-end
+```
+
+</InlineFilter>
+
+#### AMAZON_COGNITO_USER_POOLS
+
+<InlineFilter filters={["swift"]}>
+
+If you are using Amplify Auth, you can create a method that retrieves the Cognito access token
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:127]: Amplify auth

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 24]
```diff
@@ -0,0 +1,398 @@
...
+
+</InlineFilter>
+
+You can provide your own custom `fetchLatestAuthToken` provider for **AWS_LAMBDA** and **OPENID_CONNECT** auth modes. 
+
+#### AWS_IAM
+
+<InlineFilter filters={["swift"]}>
+
+If you are using Amplify Auth, you can use the following method for AWS_IAM auth
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:192]: Amplify auth

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 25]
```diff
@@ -0,0 +1,398 @@
...
+  ApolloAmplifyConnector.signAppSyncRequest(it, "us-east-1") 
+}
+//highlight-end
+```
+
+</InlineFilter>
+
+### Connecting Amplify Data to Apollo client
+
+Before you begin, you will need an Amplify Data backend deploy. To get started, see [Set up Data](/[platform]/build-a-backend/data/set-up-data/).
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:230]: Amplify data

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx (hunk 12 above)
Comment (aspittel) [reply src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:234]: missing the

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx [hunk 26]
```diff
@@ -0,0 +1,398 @@
...
+
+*  `publicAPIkey` strategy, `apiKey` authMode, **APIKeyAuthorizer**
+*  `guest` strategy, `identityPool` authMode, **IAMAuthorizer**
+*  `owner` strategy, `userPool` authMode, **AuthTokenAuthorizer**
+
+If you define multiple authorization strategies on a single model, you will have to create separate Apollo client instances for each Authorizer that you want to use in your app.
+
+### Downloading the AWS AppSync schema
+
+The schema is used by Apollo’s code generation tool to generate API code that helps you execute GraphQL operations. The following steps integrate your AppSync schema with Apollo's code generation process:
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:306]: AppSync is prefix required - always need AWS before it (AWS AppSync)

**Review by aspittel — CHANGES_REQUESTED** [review]

File: src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx (hunk 23 above)
Comment (mattcreaser) [reply src/pages/[platform]/build-a-backend/data/aws-appsync-apollo-extensions/index.mdx:127]: We use capitalized category names elsewhere in our documentation (for example, [here](https://docs.amplify.aws/android/build-a-backend/auth/set-up-auth/) and [here](https://docs.amplify.aws/android/build-a-backend/data/set-up-data/))

**Review by aspittel — APPROVED** [review]

**Pull request merged**